src/liballoc/collections/btree/node.rs

```diff
@@ -453,18 +460,25 @@ impl<'a, K, V, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
 }
 
 impl<'a, K: 'a, V: 'a, Type> NodeRef<marker::Immut<'a>, K, V, Type> {
-    fn into_key_slice(self) -> &'a [K] {
-        unsafe { slice::from_raw_parts(MaybeUninit::first_ptr(&self.as_leaf().keys), self.len()) }
+    /// SAFETY: the caller must ensure that the node has more than `idx` initialized elements.
+    unsafe fn into_key(self, idx: usize) -> &'a K {
+        debug_assert!(idx < self.len());
+        let leaf = self.as_leaf();
+        unsafe { &*MaybeUninit::first_ptr(&(*leaf).keys).add(idx) }
```

Comment:
I assume this is what you mean by "simplified"?

But now this assumes that there is no outstanding mutable reference to *any* key of this node. Maybe that's okay, but the safety comment should at least say that.

Reply:
I mean "simplified" as not relying on raw pointer code.

I'm trying to make this more clear, in addition to a comment, using:
```
 let leaf = unsafe { &(*self.as_leaf()) };
```